datafusion/optimizer/src/push_down_projection.rs

```diff
@@ -710,12 +679,11 @@ mod tests {
         let expected = "Projection: test.a, test.c, test.b\
         \n  Filter: test.a > Int32(1)\
         \n    Filter: test.b > Int32(1)\
-        \n      Filter: test.c > Int32(1)\
-        \n        TableScan: test projection=[a, b, c]";
-
-        assert_optimized_plan_eq(&plan, expected);
-
-        Ok(())
+        \n      Projection: test.c, test.a, test.b\
```

Comment:
These projections above a Filter that just pass the input to the output are also unecessary, right? Maybe we can add a rule to the remove unnecessary projections for these as well (if the schema of the projection's input is the same as the schema of its output)

Reply:
Thank you @alamb . Agree with it.
But In this example, the schema of the projection' !=  the schema of its child, because the order already changed. 
In the future, we indeed remove these projection (if it just change order, and it isn't in the top of plan tree, which means that there must be a plannode above this projection that can determine the output schema like agg, other projection ....)

Reply:
>  because the order already changed.
> In the future, we indeed can remove these projection (if it just change order, and it isn't in the top of plan tree, which means that there must be a plannode above this projection that can

That is an excellent point

> We can enhance EliminateProject Rule

Good idea 👍